From 1c3e6cfd0a095f3794b7fb989b6b86a2d18bfce7 Mon Sep 17 00:00:00 2001
From: Richard W.M. Jones <rjones@redhat.com>
Date: Wed, 29 May 2013 21:50:00 +0100
Subject: [PATCH 27/31] RHEL 6: Revert "Use pkg-config for Python"

Python in RHEL 6 does not contain the required pkg-config files.

This reverts commit e5f0dfb6e3425e8dbaa405847ae98a8dcc421da7.
---
 configure.ac       |   35 +++++++++++++++--------------------
 python/Makefile.am |    2 +-
 2 files changed, 16 insertions(+), 21 deletions(-)

diff --git a/configure.ac b/configure.ac
index 9e1d67d..d199f21 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1067,6 +1067,7 @@ AM_CONDITIONAL([HAVE_PERL],
 dnl Check for Python (optional, for Python bindings).
 PYTHON_PREFIX=
 PYTHON_VERSION=
+PYTHON_INCLUDEDIR=
 PYTHON_INSTALLDIR=
 
 AC_ARG_ENABLE([python],
@@ -1077,30 +1078,23 @@ AS_IF([test "x$enable_python" != "xno"],[
     AC_CHECK_PROG([PYTHON],[python],[python],[no])
 
     if test "x$PYTHON" != "xno"; then
+        AC_MSG_CHECKING([Python prefix])
+        PYTHON_PREFIX=`$PYTHON -c "import sys; print (sys.prefix)"`
+        AC_MSG_RESULT([$PYTHON_PREFIX])
+
 	AC_MSG_CHECKING([Python version])
         PYTHON_VERSION_MAJOR=`$PYTHON -c "import sys; print (sys.version_info@<:@0@:>@)"`
         PYTHON_VERSION_MINOR=`$PYTHON -c "import sys; print (sys.version_info@<:@1@:>@)"`
         PYTHON_VERSION="$PYTHON_VERSION_MAJOR.$PYTHON_VERSION_MINOR"
 	AC_MSG_RESULT([$PYTHON_VERSION])
-        # Debian: python-2.7.pc, python-3.2.pc
-        PKG_CHECK_MODULES([PYTHON], [python-"$PYTHON_VERSION"],[
-            AC_SUBST([PYTHON_CFLAGS])
-            AC_SUBST([PYTHON_LIBS])
-            AC_SUBST([PYTHON_VERSION])
-            AC_DEFINE([HAVE_PYTHON],[1],[Python library found at compile time])
-        ],[
-            PKG_CHECK_MODULES([PYTHON], [python],[
-                AC_SUBST([PYTHON_CFLAGS])
-                AC_SUBST([PYTHON_LIBS])
-                AC_SUBST([PYTHON_VERSION])
-                AC_DEFINE([HAVE_PYTHON],[1],[Python library found at compile time])
-            ],[
-                AC_MSG_WARN([python $PYTHON_VERSION not found])
-            ])
-        ])
-        AC_MSG_CHECKING([Python prefix])
-        PYTHON_PREFIX=`$PYTHON -c "import sys; print (sys.prefix)"`
-        AC_MSG_RESULT([$PYTHON_PREFIX])
+
+        AC_MSG_CHECKING([for Python include path])
+        if test -z "$PYTHON_INCLUDEDIR"; then
+            python_path=`$PYTHON -c "import distutils.sysconfig; \
+                                     print (distutils.sysconfig.get_python_inc ());"`
+            PYTHON_INCLUDEDIR=$python_path
+        fi
+        AC_MSG_RESULT([$PYTHON_INCLUDEDIR])
 
         AC_ARG_WITH([python-installdir],
                     [AS_HELP_STRING([--with-python-installdir],
@@ -1146,11 +1140,12 @@ AS_IF([test "x$enable_python" != "xno"],[
 
     AC_SUBST(PYTHON_PREFIX)
     AC_SUBST(PYTHON_VERSION)
+    AC_SUBST(PYTHON_INCLUDEDIR)
     AC_SUBST(PYTHON_INSTALLDIR)
     AC_SUBST(PYTHON_EXT_SUFFIX)
 ])
 AM_CONDITIONAL([HAVE_PYTHON],
-    [test "x$PYTHON" != "xno" && test "x$PYTHON_LIBS" != "x" ])
+    [test "x$PYTHON" != "xno" && test "x$PYTHON_INCLUDEDIR" != "x" && test "x$PYTHON_INSTALLDIR" != "x"])
 
 dnl Check for Ruby and rake (optional, for Ruby bindings).
 AC_ARG_ENABLE([ruby],
diff --git a/python/Makefile.am b/python/Makefile.am
index e9a9b85..2caa07f 100644
--- a/python/Makefile.am
+++ b/python/Makefile.am
@@ -40,7 +40,7 @@ libguestfsmod_la_SOURCES = guestfs-py.c guestfs-py.h guestfs-py-byhand.c
 
 libguestfsmod_la_CPPFLAGS = \
 	-DGUESTFS_PRIVATE=1 \
-	$(PYTHON_CFLAGS) \
+	-I$(PYTHON_INCLUDEDIR) \
 	-I$(top_srcdir)/src -I$(top_builddir)/src
 
 libguestfsmod_la_CFLAGS = \
-- 
1.7.1

